### DISCUSSION TITLE
No synchronisation after one bad round

### DISCUSSION THREAD

**Discussion by Plotter123** [jomjol/AI-on-the-edge-device#2737] [General]
### The Problem

Several weeks everything works fine, but today I found that since one week I have only errors. Looking in the csv file shows, that one measurement was to high (7 instaed of 1), but already the next value was good again. But this good value was marked as an error and all following values too. The reason seems to be the used timestamp to determine the rate. The timestamp to calculate the gradient is always the one of the last measured value , but as value the last correct value which of course is longer ago. So the calculated gradient is always to high and gets worser with each new measurement.

My configuration is an allowed rate of 0.04 per minute and measurements were taken every 5 minutes.

Last good measurement:     2023-11-30T07:29:48+0100,main,33890.987
One wrong measurement:   2023-11-30T07:34:48+0100,main,33897.075   five minutes later, rate too high, 1.218 correct calculation
Next measurement:             2023-11-30T07:39:49+0100,main,33891.200   ten minutes after last good measurement, marked as rate too high (0.042) but the correct rate since last good measurement is 0.021 per minute which is below my allowed limit.

### Version

15.3

### Logfile

```shell
Logfile already deleted here the csv of the interesting region

2023-11-30T07:24:48+0100,main,33890.987,33890.987,33890.987,0.002800,0.014,no error,3.0,3.0,8.0,9.0,0.0,9.0,8.2,7.3
2023-11-30T07:29:48+0100,main,33890.987,33890.987,33890.987,0.000000,0.000,no error,3.0,3.0,8.0,9.0,0.0,9.0,8.3,7.3
2023-11-30T07:34:48+0100,main,33897.075,,33890.987,,0.000,Rate too high - Read: 33897.075 - Pre: 33890.987 - Rate: 1.218,3.0,3.0,8.0,9.0,7.0,0.0,7.3,5.5
2023-11-30T07:39:49+0100,main,33891.200,,33890.987,,0.000,Rate too high - Read: 33891.200 - Pre: 33890.987 - Rate: 0.042,3.0,3.0,8.0,9.0,1.0,2.0,0.0,0.3
2023-11-30T07:44:49+0100,main,33891.267,,33890.987,,0.000,Rate too high - Read: 33891.267 - Pre: 33890.987 - Rate: 0.056,3.0,3.0,8.0,9.0,1.0,2.0,6.0,7.4
```

**Comment by friedpa** [comment]
You have to set the "Previous Value" new (in the Settings Menu)

**Comment by Plotter123** [comment]
> You have to set the "Previous Value" new (in the Settings Menu)

Yes, this is the solution if it has happened. But in my opinion the system should recover by itself.

**Comment by friedpa** [comment]
The Software is not designed like this.
E.g. The "rate too high" is a water leakage, in my case a valve is triggered to stop the water flow in the house. If the system would "recover" by itself, the valve would be opened again.... in my case: not good

**Comment by Plotter123** [comment]
I don't think that you are right. The initial message "rate too high" is due to a reading error which according to the FAQ can always happen.  My limit should filter out such impossible values and this works fine. But if next value is again in the limits (no reading error) the system should recover by itself.

**Comment by friedpa** [comment]
You can use this "feature" in both ways, look at your meter what the maximum thruput is and then measure your maximum  "normal" waterflow. Did you also check the parameter  "Maximum Age of Previous Value after Startup" 
I am using this "feature" in the aboth way, togehter with a blocky script via IOBroker.

**Comment by proschowsky** [comment]
I have the same issue with my installation. 

Documentation:
_Maximum allowed change between two readings, if exceeded the last reading will be rejected. Depending on the settings of <NUMBER>.MaxRateType the MaxRateValue is either treated as the difference between the two measurements (AbsoluteChange = difference) not taking the set time interval into account or as the difference normalized to the interval (RateChange = difference per minute)._

If the actual timestamp of the last valid measurement isn’t used to calculate rate, then this feature makes little sense.

**Comment by Plotter123** [comment]
@caco3 : Why did you remove the bug label and move the post to discussion?
